js-core/homeworks/homework-10/src/main.js

```diff
@@ -0,0 +1,138 @@
+/*
+ *
+ * Задача 0
+ *
+ * Что вернет выражение z(x) ?
+ * Напишите ответ своими словами как вы понимаете
+ * В консоле не смотрите, сначала напишите, после проверьте себя
+ *
+ * */
+
+let y = 5;
+
+//создана переменная y и ей присвоенно значение 5
+
+let x = () => y;
+
+//создана переменная x в которую записана функция, возвращающая y
+
+let z = t => {
+  let y = 5;
+  t();
+};
+//console.log(y);
+
+//созданна переменная z которая развна функции t которая создает внутреннею переменную y и вызывает сама себя
+
+console.log(z(x)); // что вернет
+
+//undefined - так как t ничего не возвращает
+
+/*
+ *
+ * TASK 1
+ * Создайте функцию которая будет превращать
+ * передаваемую строку в html тэг
+ *
+ *
+ * */
+
+let $ = tag => `<${tag}></${tag}>`;
+
+let createBODY = $("body");
+let createDIV = $("div");
+console.log(createBODY); // <body></body>
+console.log(createDIV); // <div></div>
+
+/*
+ *
+ * TASK 2
+ *
+ * Создайте объект к которому можно будет применить любое число вызовов
+  // obj.method().method().method()
+  ---------------
+ *  Передаваемое значение должно возвращаться в виде html тэгов (TASK 1)
+ *  Передаваемые аргументы должны быть только в виде строки
+ * */
+
+var ezjQueryBase = {
+  tags: "",
+  add: function(str) {
+    ezjQueryBase.tags = ezjQueryBase.tags + `<${str}></${str}>`;
+    console.log(ezjQueryBase.tags);
+    return this;
+  }
+};
+
+ezjQueryBase
+  .add("body") // <body></body>
+  .add("div") // <body></body><div></div>
+  .add("h1"); // <body></body><div></div><h1></h1>
+
+/*
+ *
+ * TASK 3
+ * Доработйте метод add чтобы на каждом вызове следующий
+ * тэг помещался внутри предыдущего !
+ ---
+ * И добавьте объекту ezjQuery метод render, который будет возвращать
+ * сгенерированную строку
+ -----
+ * Методу add - второй параметр, который будет размещать
+ * информацию внутри тэга
+ *
+ */
+var ezjQuery = {
+  tags: "",
+  add: function(...arg) {
+    let tag = [...arg][0];
+    let tagContent = [...arg][1];
+    let newTagPosition = ezjQuery.tags.indexOf("</");
```

Comment:
not sure that solution will work at all.

what if I pass "</" as second argument ? or what happens if my call would look that way
```js
ezjQuery.add('div', '<div>lallal</div>').add('main', </>__</>)
```

Have you checked that cases?